go/nbs/table_set.go

```diff
@@ -21,9 +21,9 @@ func newTableSet(persister tablePersister) tableSet {
 
 // tableSet is an immutable set of persistable chunkSources.
 type tableSet struct {
-	novel, upstream chunkSources
-	p               tablePersister
-	rl              chan struct{}
+	novel, compacted, compactees, upstream chunkSources
```

Comment:
I initially tried not to keep all the `compacted` and `compactee` state here inside `tableSet`. Instead, I had `Compact()` return that info to the caller. This was pretty nice until I got to the part where I needed that state to persist across multiple invocations to `updateManifest`, in the event that the client encountered multiple optimistic lock failures while trying to land a compaction. Essentially, you need to remember which tables are the result of a compaction, and which tables are now rolled up into a larger table allll the way until you successfully update the manifest. Trying to maintain that state outside `tableSet` turned out to be hairier than just keeping the complexity in here, though it initially seemed like it'd be cleaner and simpler.